ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -406,20 +413,63 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
                     )
 
             # L2 Vlans per Tenant
-            # If multiple L2 Vlans share the same name, they will be part of the same bundle
-            for bundle_name, l2vlans in groupby(tenant["l2vlans"], "name"):
+            # If multiple L2 Vlans share the same evpn_vlan_bundle name, they will be part of the same vlan-aware-bundle else they use the vlan name as bundle
+            sorted_vlan_list = sorted(tenant["l2vlans"], key=self._get_vlan_aware_bundle_name)
+            bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name)
+            for bundle_name, l2vlans in bundle_groups:
                 l2vlans = list(l2vlans)
-                if (bundle := self._router_bgp_vlans_vlan(l2vlans[0], tenant, vrf={})) is not None:
+                if "evpn_vlan_bundle" in l2vlans[0]:
```

Comment:
We use a groupby in line 418 to group the vlans by the result of `vlan.get("evpn_vlan_bundle")` or `vlan.get("name")` which means all vlans in that group either have` evpn_vlan_bundle` or they don't. This is why we can just check in the first element of the group

Reply:
The point is that since we use `or` a vlan without any "evpn_vlan_bundle" might still end up in the pile of vlans using the new "evpn_vlan_bundle" feature, if it shares the name of such a bundle. But in that case the settings would be taken from the "evpn_vlan_bundles" instead of the vlan, which might confuse the user.
So what Guillaume pointed out is that we might have to change this logic a bit, to check for "evpn_vlan_bundle" in all the vlans, so it is "all or nothing", and if not, raise an error.

Reply:
An alternative solution could be to prefix the name from `evpn_vlan_bundle` with `_` so they end up in different groups, and then remove the prefix again before adding to the bundle list. Our duplicate detection would then catch the overlapping names.

Reply:
Hi @bjmeuer - I appreciate my comment was a bit light just to add to what Claus said - if for some unfortunate reason someone chooses `name` for a `l2vlan` that is clashing with you bundle name (I know my example looks a but simple) - then the order of the l2vlans will have an impact with current code (because `l2vlans[0]` won't have an `evpn_vlan_bundle` but WILL have been grouped indeed with other vlans, because of its name

```
---
  - name: FABRIC
    vlan_aware_bundle_number_base: 200
    mac_vrf_vni_base: 10000
    l2vlans:
      - id: 50
        name: bundle1 <<<<<<<<<<<<<<<<<<<<<<<<<< unfortunate vlan name without evpn_vlan_bundle
        tags: ['ipmi']
      - id: 60
        name: ipmi
        tags: ['fabric']
        evpn_vlan_bundle: bundle1

evpn_vlan_bundles:
  - name: bundle1
    id: 1
    bgp:
      raw_eos_cli: |
        vlan add 1000
```

Reply:
change to adding a temporary "_" like Claus suggested

Reply:
the last change does not solve the issue if the vlan name is `_bundle1`

Reply:
ok, think if I temporary prefix the vlan name as well with a different character this should work right?

Reply:
I have implemented a different fix for this, which is more well defined instead of hacking prefixes.

Reply:
This code is so much better. I was focusing on that prefix thing too much so I didn't thought of that...